activerecord/lib/active_record/reflection.rb

```diff
@@ -913,11 +915,13 @@ def source_reflection_names
       def source_reflection_name # :nodoc:
         return @source_reflection_name if @source_reflection_name
 
-        names = [name.to_s.singularize, name].collect(&:to_sym).uniq
```

Comment:
If this change is safe, I think the whole code block is eliminated -- if we start with a single-element array, then after uniq and filtering it's definitely not going to end up with multiple elements.

But I'm not sure the change is safe (yet?): the change to `_reflect_on_association` will find plural-named associations when given a singular name, but here we're removing a singularize... if `name` here is plural, after this change we'd no longer find a singular version AFAICS.

Reply:
Yeah, I think I was wrong to change this and we still want to make sure that in `_reflections` we don't have ambiguous names.